### ISSUE TITLE
lua-source cannot work in batch mode

### ISSUE THREAD

**Issue by Freed-Wu** [neomutt/neomutt#4328]
## Expected Behaviour

```
neomutt -F /dev/null -B -e 'lua-source /dev/shm/a.lua'
```

Work.

## Actual Behaviour

Couldn't source lua source: cannot open : No such file or directory
Error in command line:

If NeoMutt crashed, did your OS create a 'coredump' file?

No

## Steps to Reproduce

```
neomutt -F /dev/null -B -e 'lua-source /dev/shm/a.lua'
```

## How often does this happen?

- Always

## When did it start to happen?

- When I found https://neomutt.org/2017/04/29/lua

## NeoMutt Version

```
NeoMutt 20240425
Copyright (C) 2015-2024 Richard Russon and friends
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Linux 6.9.1-arch1-1 (x86_64)
ncurses: ncurses 6.4.20230520 (compiled with 6.4.20230520)
libidn2: 2.3.7 (compiled with 2.3.7)
GPGME: 1.23.2
GnuTLS: 3.8.5
libnotmuch: 5.6.0
storage: kyotocabinet, gdbm, lmdb
compression: lz4, zlib, zstd

Configure options: --prefix=/usr --sysconfdir=/etc --libexecdir=/usr/lib --gpgme --sqlite --autocrypt --lua --notmuch --gss --gnutls --sasl --with-ui=ncurses --with-idn2=/usr --disable-idn --idn2 --lmdb --kyotocabinet --gdbm --ubsan --lz4 --zlib --zstd

Compilation CFLAGS: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -g -ffile-prefix-map=/build/neomutt/src=/usr/src/debug/neomutt -flto=auto -fno-delete-null-pointer-checks -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -D_XOPEN_SOURCE_EXTENDED -I/usr/include/lua5.3 -I/usr/include -I/usr/include -I/usr/include -DNCURSES_WIDECHAR -I/usr/include -I/usr/include/p11-kit-1 -I/usr/include -I/usr/include -fsanitize=undefined -fno-omit-frame-pointer -fno-common -g -O0

Compile options:
  +autocrypt +fcntl -flock -fmemopen +futimens +getaddrinfo +gnutls +gpgme
  -gsasl +gss +hcache -homespool +idn +inotify -locales_hack +lua -mixmaster
  +nls +notmuch -openssl +pgp +regex +sasl +smime +sqlite +truecolor

Devel options:
  ubsan

MAILPATH="/var/mail"
PKGDATADIR="/usr/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

## Extra Info

* Operating System and its version
```
uname (GNU coreutils) 9.5
Copyright (C) 2024 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by David MacKenzie.
```
* Were you using multiple copies of NeoMutt at once?
No
* Were you using 'screen' or 'tmux'?
tmux
* Is your email local (maildir) or remote (IMAP)?
remote

**Comment by roccoblues** [comment]
Hi, sorry about that.

"The Lua support is still very immature" (quoting from https://neomutt.org/2017/04/29/lua). Unfortunately 7 years later this is still very much true because no one worked on it.

tbh, I'd be in favor of removing instead of fixing it. Imho the costs far outweight the potential benefits.

**Comment by gahr** [comment]
> tbh, I'd be in favor of removing instead of fixing it. Imho the costs far outweight the potential benefits.

I agree - it's never been finished and it shouldn't have been released in my opinion. But we don't know how many users are using it now, so I don't think removing it is an option anymore, alas.

**Comment by gahr** [comment]
I'm all for removing features *I* don't use :)

**Comment by flatcap** [comment]
I'd _prefer_ **not** to kill lua.
Scripting has potential -- think of the difference between a text editor and Vim.

Keeping nntp is harder for me to justify.
It _does_ work, but other NeoMutt upgrades have left it very poorly integrated.

**Comment by jindraj** [comment]
I'd prefer to not kill lua too.

Lua doesn't offer much features now. It has poor user adoption, but there are really helpful usecases. E.g. Notmuch time windoes are really cool, but controlling `nm_query_window_*` variables is pain. I have macros for shifting windows forwards/backwards, increment/decrement nm_query_window_duration, but changing nm_query_window_timebase` without manually type the set co[here](https://github.com/jindraj/dotfiles/blob/master/.neomutt/lua/rotate_window_timebase.lua) I have a [macro](https://github.com/jindraj/dotfiles/blob/master/.neomutt/neomuttrc.bindings#L144-L146) that changes a variable

It would be nice to be able to get headers or body in the future.

**Comment by jindraj** [comment]
I'd prefer to not kill lua too.

Lua doesn't offer much features now. It has poor user adoption, but there are really helpful usecases. E.g. Notmuch time windows are really cool, but controlling `nm_query_window_*` variables is pain. I have macros for shifting windows forwards/backwards, increment/decrement `nm_query_window_duration`, but for changing `nm_query_window_timebase` i'd need to manually type the set command. [Here](https://github.com/jindraj/dotfiles/blob/master/.neomutt/lua/rotate_window_timebase.lua) comes lua to scene - I have a [macro](https://github.com/jindraj/dotfiles/blob/master/.neomutt/neomuttrc.bindings#L144-L146) that changes that variable. 

It would be nice to be able to get headers or body in the future.

**Comment by gahr** [comment]
Cool so what do we do with this? Are you still interested in enhancing the support for Lua, @jindraj ? Or is anybody else? Or should we close all reported limitations as "won't fix"?

**Comment by jindraj** [comment]
Unfortunately, I'm afraid this is far beyond my skills. If there's not currently anyone to improve lua support, I'd lean to close all reported limitations as „won't fix“ keep lua marked as experimental and wait until its time comes.

**Comment by Freed-Wu** [comment]
```sh
$ neomutt -B
Invalid option: B
```

NeoMutt 20250510-dirty doesn't have `-B` so this bug cannot be reproduced. closed.
